apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -116,6 +116,10 @@
 
 -callback open_db(db(), create_db_opts()) -> ok | {error, _}.
 
+-callback update_db_config(db(), create_db_opts()) -> ok | {error, _}.
+
+-callback add_generation(db()) -> ok | {error, _}.
```

Comment:
Maybe we leave just `add_generation` with opts? Or `add_generation` with and without opts.

Otherwise the api may be a bit confusing:
* if we want to add gen changing opts, then we should use `update_db_config`
* if we want to add gen keeping opts, then we should use `add_generation`

Reply:
Yes,  them can be combined by `add_generation/0`, `add_generation/1`,  @ieQu1 does such modification meet our goals?